src/shogun/classifier/svm/NewtonSVM.cpp

```diff
@@ -197,34 +196,22 @@ bool CNewtonSVM::train_machine(CFeatures* data)
 		SG_FREE(Xsv2sum);
 		SG_FREE(identity_matrix);
 		SG_FREE(inverse);
-		SG_FREE(step);
 		SG_FREE(s2);
 
 		if (newton_decrement*2<prec*obj)
-			break;
+		m_complete=true;
+}
 
-		pb.print_progress();
-	}
-	pb.complete();
+void CNewtonSVM::end_training()
```

Comment:
i think this method can be removed if it doesnt do anything or?

Reply:
yeah it is not needed anymore